File: 0_array/question/removeDuplicates/w1.cpp
#include <climits>
#include <iostream>
#include <queue>
#include <stack>
#include <vector>
 #include <map>
using namespace std;
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int j = 0;
        int n = nums.size();
        for(int i = 0 ; i < n ; i++){
            nums[j] = nums[i];
            j++;
            while(i <  n  && nums[i] == nums[i+1] ){
                i++;
            }
        }
        return j;
    }
};

File: 0_array/question/removeDuplicates/w1.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function (nums) {
  let i = 0
  let c = 0
  while (i < nums.length) {
    while (nums[i] == nums[i + 1]) {
      i++
    }
    ;[nums[i], nums[c]] = [nums[c], nums[i]]
    c++
    i++
  }
  return c
}
